File: Customers-Products-Orders/order-service/src/main/java/com/benyissa/orderservice/entities/Order.java
package com.benyissa.orderservice.entities;


import com.benyissa.orderservice.enums.Status;
import com.benyissa.orderservice.models.Customer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "orders")
@Builder
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    Status status;
    Long costumerId;
    Date createdAt;
    Date updatedAt;
    @OneToMany(mappedBy = "order")
    List<ProductItem> orderedProducts;

    // we can do that using dto , this annotation tells JPA please don't care about this attribute
    @Transient
    Customer customer;
}

File: Customers-Products-Orders/frontend/src/app/models/Order.ts
import { Product } from './Product';

export interface OrderServiceResponse {
  _embedded: OrderEmbedded;
}

export interface OrderEmbedded {
  orders: Order[];
}

export interface Order {
  id: number;
  status: string;
  costumerId: number;
  updatedAt: Date;
  createdAt: Date;
  total: number;
  orderedProducts: OrderedProducts[];
}

export interface OrderedProducts {
  id: number;
  productId: number;
  quantity: number;
  discount: number;
  product: Product;
  amount: number;
}
